src/components/tipRecords/TipRecord.vue

```diff
@@ -206,45 +210,77 @@ export default {
       return isPinned !== -1;
     },
   },
+  mounted() {
+    if (!this.$route.path.includes(`/tip/${this.tip.id}`)) {
+      return;
+    }
+    const { method, challenge, signature } = this.$route.query;
+    if (!this.useSdkWallet && method && challenge && signature) {
+      this.applyBackendChanges(method, { challenge, signature });
+    }
+  },
   methods: {
     ...mapMutations(['setPinnedItems']),
-    async sendReport() {
-      Backend.sendPostReport(
-        this.tip.id,
-        client.rpcClient.getCurrentAccount(),
-        (data) => client.signMessage(data),
-      ).then(() => {
-        this.showSuccessModal = true;
-      }).catch((error) => {
-        console.log(error);
-      });
-    },
     async claim() {
       const postData = {
         url: this.tip.url,
         address: client.rpcClient.getCurrentAccount(),
       };
       await Backend.claimFromUrl(postData);
     },
-    pinOrUnPinTip(pinTip = true) {
-      if (!this.account) {
-        return;
+    async applyBackendChanges(method, request) {
+      const args = {
+        pinItem: [this.account, request],
+        unPinItem: [this.account, request],
+        sendPostReport: [request],
+      }[method];
+
+      if (!args) throw new Error(`Unknown method: ${method}`);
+      await Backend[method](...args);
+
+      if (method === 'pinItem' || method === 'unPinItem') {
+        const updatedPinnedItems = await Backend.getPinnedItems(this.account);
+        this.$store.commit('setPinnedItems', updatedPinnedItems);
+      }
+
+      if (method === 'sendPostReport') {
+        this.showSuccessModal = true;
+      }
+    },
+    async backendAuth(method, challenge) {
+      if (this.useSdkWallet) {
+        const signature = await client.signMessage(challenge);
+        this.applyBackendChanges(method, { challenge, signature });
+      } else {
+        const url = new URL(`${window.location.origin}/tip/${this.tip.id}`);
+        url.search = '';
+        window.location = createDeepLinkUrl({
+          type: 'sign-message',
+          message: challenge,
+          'x-success': `${url}?method=${method}&challenge=${challenge}&signature={signature}`,
+        });
+      }
+    },
```

Comment:
This code looks similar to one in user profile: https://github.com/aeternity/superhero-ui/blob/37d0eb7ce88f042dd13625345ab2e26301169d9b/src/views/UserProfile.vue#L369-L390
I propose to extract it somehow